addons/mail/controllers/main.py

```diff
@@ -115,11 +115,11 @@ def mail_action_view(self, model=None, res_id=None, message_id=None):
 
         # find the access action using sudo to have the details about the access link
         RecordModel = request.env[model]
-        record_sudo = RecordModel.sudo().browse(res_id).exists()
-        if not record_sudo:
+        record = RecordModel.browse(res_id).exists()
```

Comment:
I do not really understand why here you are able browse the record without sudo
while in `get_formview_id`, `get_access_action`, ... you can't.

Besides, would this work if you receive a mail with a button linking to the record, but you loss the access after the sending of the mail (like you have been removed from the followers, because you were added by mistake ?)